src/utils/data.js

```diff
@@ -71,6 +87,12 @@ export async function fetchDataForRouteEnter(to, from, next) {
       throw false;
     }
 
+    if (error instanceof RedirectError) {
+      // throw the redirect location, so it's passed to the `next` error handler and
+      // vue router redirects to that location
+      throw transformDataPathToRoutePath(error.location);
```

Comment:
I'm guessing you already tried this and it didn't work, but could we just explicitly call `next(error.location)` here instead of rethrowing the string and relying on the navigation guard `catch` to call `next`?

Reply:
Er, I mean `next(transformDataPathToRoutePath(error.location))`

Reply:
It will work, but the promise from `fetchDataForRouteEnter` would resolve and because we return nothing, for a brief moment, we would get an empty page, because `vm.topicData = undefined` basically. 

We can also return the JSON and call next, then in the `beforeRouteUpdate` somehow detect what happened and skip fetching, but at this point it feels overcomplicated.